### ISSUE TITLE
Builds failing due tslint crash using nightly build

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#6478]
Looks like it was introduced by 9df1ed4dd32408fcaa92d031d1f3ff6468506b95

Repro

1. From an initialized TypeScript repo (i.e. you've already run `npm install` within it), run `npm install typescript@1.8.0-dev.20160114`
2. Run `jake lint`.

Expected: No lint errors.
Actual:

```shell
$ jake lint
jake aborted.
TypeError: Cannot read property 'name' of undefined
    at getPropertySymbolsFromBaseTypes (C:\Users\drosen\TypeScript2\node_modules\typescript\lib\typescript.js:49945:73)
    at getPropertySymbolFromTypeReference (C:\Users\drosen\TypeScript2\node_modules\typescript\lib\typescript.js:49970:29)
```

@yuit can you look into this?

Any fix should also include a revert of 62c3bfb1fb6714fc383379caae8b55d81bf62fbe from #6484.

**Comment by DanielRosenwasser** [comment]
The fix is pretty trivial. Just remove the `symbol &&` in `getPropertySymbolsFromBaseTypes` and add a guard to the top like

```ts
if (!symbol) {
    return;
}
```

I just don't know what a solid repro is.

**Comment by DickvdBrink** [comment]
CC: @adidahiya @JKillian  - you might be interested as well

**Comment by DanielRosenwasser** [comment]
If it helps, it seems to occur when linting `src/compiler/utilities.ts` and calling `getPropertySymbolsFromBaseTypes` with `"getCanonicalFileName"` as a property name.

**Comment by JKillian** [comment]
Thanks for the heads-up @DickvdBrink. If anything needs changing/investigating in TSLint I'll do it ASAP.

**Comment by JKillian** [comment]
Made the change suggested by @DanielRosenwasser and added a test case over in https://github.com/Microsoft/TypeScript/pull/6507.

I reverted https://github.com/Microsoft/TypeScript/commit/62c3bfb1fb6714fc383379caae8b55d81bf62fbe as requested as well, but of course that will make the CI fail for the PR, as the CI is now using the previous nightly which still contains the bug.

**Comment by DanielRosenwasser** [comment]
Thanks a ton @JKillian we really appreciate it. I'll perform the revert tonight/tomorrow morning.
